geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert2/transforms/ExpressionParser.scala

```diff
@@ -67,7 +67,7 @@ private [transforms] class ExpressionParser extends BasicParser {
   }
 
   private def field: Rule1[Expression] = rule("$field") {
-    "$" ~ unquotedString ~~> { i => FieldLookup(i) }
+    "$" ~ (unquotedString | "{" ~ oneOrMore(char | "." | " ") ~> { c => c } ~ "}") ~~> { name => FieldLookup(name) }
```

Comment:
Ok, time to ask about corner cases....

the Rule/Expression  "{" ~ oneOrMore(char | "." | " ") ~> { c => c } ~ "}" would match input like 

"{.}"
"{ . }"
"{foo...bar}"
"{ foo . bar }"
right?  

Do we want to be that permissive?  Am I missing a part of the syntax that'd avoid really weird things like that?

Reply:
Oh, also, we could have multiple spaces and multiple dots.  Is that going to work every where with GeoTools/OGC?  

Or is this just inside the GeoMesa converters?

Reply:
this is just inside the geomesa converters. yeah, you could use `${.}` and it would parse ok, but then presumably it would fail with an error that the field `.` doesn't exist... i could make it check for stricter syntax but seems like it should be handled by checks later on.

Reply:
Cool.  Just makings sure I understand.  

If people start hurting themselves on this, we can revisit:).